src/node_buffer.cc

```diff
@@ -541,6 +659,121 @@ void WriteDoubleBE(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+template <typename T, enum Endianness endianness,
+         T (*strtoT)(const char*, char**, int)>
+uint32_t WriteInt64Generic(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsObject());
+  ARGS_THIS(args[0].As<Object>())
+
+  T val;
+  if (args[1]->IsNumber()) {
+    val = args[1]->IntegerValue();
+  } else if (args[1]->IsString()) {
+    // Have to do this because strt(u)oll doesn't set errno to 0 on success
+    errno = 0;
+    node::Utf8Value str(env->isolate(), args[1]);
+    val = strtoT(*str, nullptr, 0);
+
+    if (errno != 0) {
+      if (errno == EINVAL) {
+        env->ThrowTypeError("value is invalid");
+      } else if (errno == ERANGE) {
```

Comment:
Well, since the string is supposed to be a 64 bit number (base 10) then seems we could get around the `ERANGE` error by simply checking `str.length() <= 16`. Unless it's possible that a string that small could still return `ERANGE`.